portal-api/src/v1/users/routes.test.js

```diff
@@ -90,4 +101,103 @@ describe('users routes', () => {
       );
     });
   });
+
+  describe('validateSignInLink', () => {
+    const signInToken = 'signInToken';
+    const user = {
+      _id: new ObjectId(),
+      username: 'HSBC-maker-1',
+      firstname: 'Mister',
+      surname: 'One',
+      email: 'one@email.com',
+      timezone: 'Europe/London',
+      roles: [MAKER],
+      bank: {
+        id: '961',
+        name: 'HSBC',
+        emails: ['maker1@ukexportfinance.gov.uk', 'maker2@ukexportfinance.gov.uk'],
+      },
+    };
+
+    const req = {
+      params: { signInToken },
+      user
+    };
+
+    const sessionToken = 'sessionToken';
+    const expires = '12h';
+
+    const validateSignInLinkTokenResponse = {
+      tokenObject: {
+        token: sessionToken,
+        expires,
+      },
+      user
+    };
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it('returns a 200 response with the correct response body if the sign in link token is valid', async () => {
+      const res = {
+        status: jest.fn().mockReturnThis(),
+        json: jest.fn(() => 'mockResponse'),
+      };
+
+      when(validateSignInLinkToken).calledWith(user, signInToken).mockResolvedValue(validateSignInLinkTokenResponse);
+
+      const mockResponse = await validateSignInLink(req, res);
+
+      expect(res.status).toHaveBeenCalledTimes(1);
+      expect(res.status).toHaveBeenCalledWith(200);
+      expect(res.json).toHaveBeenCalledTimes(1);
+      expect(res.json).toHaveBeenCalledWith({
+        success: true,
+        token: sessionToken,
+        user,
+        loginStatus: LOGIN_STATUSES.VALID_2FA,
+        expiresIn: expires,
+      });
+      expect(mockResponse).toBe('mockResponse');
+    });
+
+    it('returns a 410 response with the correct response body if validating the sign in link token throws a SignInLinkExpiredError', async () => {
+      const res = {
+        status: jest.fn().mockReturnThis(),
+        json: jest.fn(() => 'mockResponse'),
+      };
+
+      when(validateSignInLinkToken).calledWith(user, signInToken).mockRejectedValue(new SignInLinkExpiredError());
+
+      const mockResponse = await validateSignInLink(req, res);
+
+      expect(res.status).toHaveBeenCalledTimes(1);
+      expect(res.status).toHaveBeenCalledWith(410);
+      expect(res.json).toHaveBeenCalledTimes(1);
+      expect(res.json).toHaveBeenCalledWith({
+        success: false,
+      });
+      expect(mockResponse).toBe('mockResponse');
+    });
+
+    it('returns a 410 response with the correct response body if validating the sign in link token throws an error other than a SignInLinkExpiredError', async () => {
```

Comment:
```suggestion
    it('returns a 500 response with the correct response body if validating the sign in link token throws an error other than a SignInLinkExpiredError', async () => {
```

Reply:
Good spot, but this logic has moved/changed now anyway.